### PR TITLE
Update dependency on Trusted Services

### PR DESCRIPTION
**Pull request by ionut-arm**
This commit updates our dependency on the upstream Trusted Services repo
to include recent commits. The changes include major updates to the
crypto interface, including a move to PSA Crypto v1, meaning a very
different approach to managing keys is needed. Calls to OpenKey and
CloseKey have been scrapped.

### CODE REVIEW

File: src/providers/trusted_service/context/mod.rs [hunk 1]
```diff
@@ -10,7 +10,10 @@ use std::ptr::null_mut;
 use std::slice;
 use std::sync::Mutex;
 use ts_binding::*;
-use ts_protobuf::{CloseKeyIn, GetOpcode, OpenKeyIn, OpenKeyOut, SetHandle};
+use ts_protobuf::GetOpcode;
+
+// Bindgen fails to generate this constant.
+pub const PSA_KEY_ID_USER_MAX: u32 = 0x3fffffff;
```
Comment (hug-dev) [inline src/providers/trusted_service/context/mod.rs:16]: I don't know if you have access but it's also [here](https://docs.rs/psa-crypto/0.8.0/psa_crypto/types/key/constant.PSA_KEY_ID_USER_MAX.html)

File: e2e_tests/docker_image/parsec-service-test-all.Dockerfile [hunk 2]
```diff
@@ -94,15 +94,18 @@ RUN cd nanopb-0.4.4-linux-x86 \
 RUN rm -rf nanopb-0.4.4-linux-x86 nanopb-0.4.4-linux-x86.tar.gz
 
 # Install mock Trusted Services
-RUN git clone https://git.trustedfirmware.org/TS/trusted-services.git --branch main \
+# Setup git config for patching dependencies
+RUN git config --global user.email "some@email.com"
+RUN git config --global user.name "Parsec Team"
```
Comment (hug-dev) [inline e2e_tests/docker_image/parsec-service-test-all.Dockerfile:99]: Fro what commands is it needed?

**Review by hug-dev — APPROVED** [review]
Nice 👌

File: e2e_tests/docker_image/parsec-service-test-all.Dockerfile (hunk 2 above)
Comment (hug-dev) [reply e2e_tests/docker_image/parsec-service-test-all.Dockerfile:99]: Ignore my message, answer [here](https://github.com/parallaxsecond/parsec/pull/406#discussion_r666309798)

File: src/providers/trusted_service/context/mod.rs (hunk 1 above)
Comment (ionut-arm) [reply src/providers/trusted_service/context/mod.rs:16]: Ah, yes, nice!

**Review by hug-dev — APPROVED** [review]

**Pull request merged**